service/matching/matchingEngine.go

```diff
@@ -380,21 +380,21 @@ func (e *matchingEngineImpl) DispatchSpooledTask(
 	ctx context.Context,
 	task *internalTask,
 	origTaskQueue *taskQueueID,
+	kind enumspb.TaskQueueKind,
 ) error {
 	// This task came from taskReader so task.event is always set here.
 	directive := task.event.GetData().GetVersionDirective()
 	// If this came from a versioned queue, ignore the version and re-resolve, in case we're
 	// going to the default and the default changed.
 	unversionedOrigTaskQueue := newTaskQueueIDWithVersionSet(origTaskQueue, "")
-	// Kind must be normal here, sticky aren't spooled.
-	kind := enumspb.TASK_QUEUE_KIND_NORMAL
 	// Redirect and re-resolve if we're blocked in matcher and user data changes.
 	for {
 		taskQueue, userDataChanged, err := e.redirectToVersionedQueueForAdd(ctx, unversionedOrigTaskQueue, directive, kind)
 		if err != nil {
 			return err
 		}
-		tqm, err := e.getTaskQueueManager(ctx, taskQueue, kind, true)
+		sticky := kind == enumspb.TASK_QUEUE_KIND_STICKY
+		tqm, err := e.getTaskQueueManager(ctx, taskQueue, kind, !sticky)
```

Comment:
Passing sticky seems redundant with kind, but, you didn't change that so fine to just leave it

Reply:
@dnr

Reply:
Most of the time they are redundant but there are a few situations where it does something different